GTBitsOfGood/canine-assistants
# Edit Individual Dog Page

## Edit Individual Dog Page
Issue Number(s): #31 

What does this PR change and why?
- will allow editing of each dog field with zod validation

### Acceptance Criteria
- [x] text/display fields are converted to input fields when editing
- [x] client-side validation is implemented
- [x] `PATCH` request is sent to `/api/dogs/:id` with the appropriate request body
- [x] proper error handling including displaying success and failure to the client via "toasts"
- [x] proper state management (the dog on the client should always match the dog data on the server)


### Things to do
- fields that are N/A currently don't update (pushing over to next sprint due to intricacies) 

### How To Test
- click edit button on dog page
- change some values 
-  submit form
- toast should be shown

### Testing Error Cases
- set Sex to value other than "Male" or "Female", validation error should arise. Try submitting (program should not submit), try hitting cancel (program should reset values back to what it was)
- set Date to an invalid date value and try the above
